daemon/logger/journald/read.go

```diff
@@ -299,7 +299,11 @@ func (s *journald) followJournal(logWatcher *logger.LogWatcher, j *C.sd_journal,
 	// Wait until we're told to stop.
 	select {
 	case cursor = <-newCursor:
-	case <-logWatcher.WatchClose():
+	case <-logWatcher.WatchProducerGone():
+		// Notify the other goroutine that its work is done.
+		C.close(pfd[1])
+		cursor = <-newCursor
```

Comment:
Wouldn't we still want to send the rest of the logs?

Reply:
this just mimics the old behavior (and minimizes the patch impact); as I mentioned in commit message:

> 2. Most of the code waiting for WatchClose() is modified to wait for
> either ConsumerGone() or ProducerGone(), which mimics the old behavior,
> except for a few places where it's perfectly clear which one we
> are to wait for. TODO: further specification for some cases might
> be needed to fix other bugs.

so yes, you're probably right but this can be addressed later